python/tchannel/tornado/stream.py

```diff
@@ -71,7 +77,7 @@ def write(self, chunk):
         if self._state == StreamState.completed:
             raise StreamingException("Stream has been closed.")
 
-        self._stream += chunk
+        self._stream.append(chunk)
```

Comment:
In both appends, you probably want to make sure that you append only if the string is non-empty.